sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchServiceClient.java

```diff
@@ -83,15 +85,18 @@ public DataSource createOrUpdateDataSource(DataSource dataSource) {
      * Creates a new Azure Cognitive Search data source or updates a data source if it already exists.
      *
      * @param dataSource the definition of the data source to create or update
-     * @param accessCondition the condition where the operation will be performed if the ETag on the server matches or
+     * @param onlyIfUnchanged the condition where the operation will be performed if the ETag on the server matches or
      * doesn't match specified values
      * @param requestOptions additional parameters for the operation. Contains the tracking ID sent with the request to
      * help with debugging
      * @param context additional context that is passed through the HTTP pipeline during the service call
      * @return a response containing data source that was created or updated.
      */
-    public Response<DataSource> createOrUpdateDataSourceWithResponse(DataSource dataSource,
-        MatchConditions accessCondition, RequestOptions requestOptions, Context context) {
+    public Response<DataSource> createOrUpdateDataSourceWithResponse(DataSource dataSource, Boolean onlyIfUnchanged,
+        RequestOptions requestOptions, Context context) {
+        MatchConditions accessCondition = new MatchConditions();
+        accessCondition = onlyIfUnchanged != null && onlyIfUnchanged ?
+            accessCondition.setIfMatch(dataSource.getETag()) : accessCondition;
         return asyncClient.createOrUpdateDataSourceWithResponse(dataSource, accessCondition, requestOptions, context)
```

Comment:
Let's just update the internal APIs to take the `onlyIfUnchanged` boolean instead of adding the logic twice.

Reply:
I am hoping service team will change the swagger sooner. 
We need to change to ifMatch and ifNoneMatch to match the headers.

Reply:
Will remove from transformer and clean up MatchConditions related changes.